sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGenerateExec.scala

```diff
@@ -717,65 +719,97 @@ case class GpuGenerateExec(
           GpuBindReferences.bindGpuReferences(requiredChildOutput, child.output)
 
         child.executeColumnar().flatMap { inputFromChild =>
-          withResource(inputFromChild) { input =>
-            doGenerate(input, genProjectList, othersProjectList,
-              numOutputRows, numOutputBatches, opTime)
-          }
+          doGenerateAndClose(inputFromChild, genProjectList, othersProjectList,
+            numOutputRows, numOutputBatches, opTime)
         }
     }
   }
 
-  private def doGenerate(input: ColumnarBatch,
+  private def doGenerateAndClose(input: ColumnarBatch,
       genProjectList: Seq[GpuExpression],
       othersProjectList: Seq[GpuExpression],
       numOutputRows: GpuMetric,
       numOutputBatches: GpuMetric,
       opTime: GpuMetric): Iterator[ColumnarBatch] = {
-    withResource(new NvtxWithMetrics("GpuGenerateExec", NvtxColor.PURPLE, opTime)) { _ =>
+    val splits = withResource(new NvtxWithMetrics("GpuGenerateExec",
+      NvtxColor.PURPLE, opTime)) { _ =>
       // Project input columns, setting other columns ahead of generator's input columns.
       // With the projected batches and an offset, generators can extract input columns or
       // other required columns separately.
-      val projectedInput = GpuProjectExec.project(input, othersProjectList ++ genProjectList)
-      withResource(projectedInput) { projIn =>
-        // 1. compute split indices of input batch
-        val splitIndices = generator.inputSplitIndices(
-          projIn,
-          othersProjectList.length,
-          outer,
-          new RapidsConf(conf).gpuTargetBatchSizeBytes)
-        // 2. split up input batch with indices
-        makeSplitIterator(projIn, splitIndices).map { splitIn =>
-          withResource(splitIn) { splitIn =>
-            // 3. apply generation on each (sub)batch
-            val ret = generator.generate(splitIn, othersProjectList.length, outer)
-            numOutputBatches += 1
-            numOutputRows += ret.numRows()
-            ret
-          }
-        }
-      }
+      val projectedInput = GpuProjectExec.projectAndCloseWithRetrySingleBatch(
+        SpillableColumnarBatch(input, SpillPriorities.ACTIVE_ON_DECK_PRIORITY),
```

Comment:
I meant to change the spill priority to ACTIVE_BATCHING_PRIORITY here and below.  Do you agree that would be more appropriate?

Reply:
Priorities are not followed all of the time consistently. I would keep as ACTIVE_ON_DECK_PRIORITY here, because it is a short lived spillable, and you want to avoid spilling here if possible..

Reply:
agree